### ISSUE TITLE
dice

### ISSUE THREAD

**Issue by tricksterANDthug** [282857341/nnFormer#71]
我想问你们代码中的acdc数据集中计算dice损失，一般我看dice损失有两种计算方式，一种是通过混淆矩阵计算tp fp tn fn等，一种是计算交并比，就是这种方式
def dice_loss(target，predictive，ep=1e-8):
        intersection = 2 * torch.sum(predictive * target) + ep
        union = torch。sum(predictive) + torch.sum(target) + ep
        loss = 1 - intersection / union
        return loss
但我感觉你们的是这两种都借鉴了吗？混合体吗

**Comment by 282857341** [comment]
不好意思这位朋友，我没有弄清楚你的问题。你的意思是询问我们计算dice的方式还是计算dice loss的方式呢。
如果是dice loss，我们的计算方式就是利用tp,fp,tn,fn计算的，代码如下：
https://github.com/282857341/nnFormer/blob/02981008c99c7be6fad004f6a0b9e237d6985707/nnformer/training/loss_functions/dice_loss.py#L158

如果是dice的计算方式，代码如下。计算dice的时候是不需要像计算dice loss那样进行1-操作的，在我的视角里，您可能有些误解。
https://github.com/282857341/nnFormer/blob/02981008c99c7be6fad004f6a0b9e237d6985707/nnformer/inference_acdc.py#L16
